messages/proposed_event.go

```diff
@@ -1,14 +1,108 @@
 package messages
 
 import (
+	"encoding/json"
+
 	uuid "github.com/gofrs/uuid"
 )
 
 // ProposedEvent ...
 type ProposedEvent struct {
-	EventID      uuid.UUID
-	EventType    string
-	ContentType  string
-	Data         []byte
-	UserMetadata []byte
+	eventID      uuid.UUID
+	eventType    string
+	contentType  string
+	data         []byte
+	metadata []byte
+}
+
+func NewJsonProposedEvent(eventType string, payload interface{}) (ProposedEvent, error) {
+	event := ProposedEvent {}
+	bytes, err := json.Marshal(payload)
+
+	if err != nil {
+		return event, err
+	}
+
+	event.eventType = eventType
+	event.contentType = "application/json"
```

Comment:
I guess that json is not the only valid way of marshaling events? As I understand, using json is giving some advantages (for example building projections on the level of EventStore DB). But I guess that it may be helpful to have some some flexibility here and allow people to use whatever format they want.

Maybe it will be better to keep  `ProposedEvent` fields public?

Reply:
(BTW it would solve some next comments as well)

Reply:
It's either `application/json` or `application/octet-stream`. I'm open to make those fields public if it's the way to go

Reply:
I think that the best option may be to have these fields public + helpers that create struct with application/json or application/octet-stream. It's giving the most flexibility and nice API.

Reply:
Done